packages/main/src/index.ts

```diff
@@ -37,6 +40,53 @@ function createWindow() {
   })
 }
 
+function handleContentDownloads() {
+  protocol.registerStreamProtocol("body", async (request, callback) => {
+    try {
+      const parsed = parseBodyUri(request.url)
+      if (!parsed) {
+        throw new Error(
+          `Unable to parse messageId and partId from URI: ${request.url}`
+        )
+      }
+      const messageId = parsed.messageId
+      const partId = parsed.partId
+      const buffer = cache.getBody(messageId, { part_id: partId })
+      const stream = createStream(buffer)
+      const part = cache.getPartByPartId({ messageId, partId })
+      const imapPart = part && cache.toImapMessagePart(part)
+      const type = imapPart && contentType(imapPart)
+      const file = imapPart && filename(imapPart)
+      callback({
+        statusCode: 200,
+        headers: {
+          "content-type": type,
+          "content-disposition": `attachment; filename= ${file}`
+        },
+        data: stream
+      })
+      stream.resume()
+    } catch (err) {
+      console.error("error serving part content:", err)
+      callback({
+        statusCode: 500,
+        headers: {
+          "content-type": "text/plain; charset=utf8"
+        },
+        data: createStream(err.message)
```

Comment:
It's not necessary for this feature, but it could be a useful exercise to send specific error codes for specific problems. That would be `404 Not Found` if `imapPart` is not defined, `400 Bad Request` if `parsed` is not defined, and `500` if we catch any other exception.